eogrow/pipelines/rasterize.py

```diff
@@ -79,45 +43,82 @@ class Schema(Pipeline.Schema):
         )
-        columns: List[VectorColumnSchema]
-        preprocess_dataset: Optional[Preprocessing] = Field(
+        raster_value: Optional[float] = Field(
+            description="Value to be used for all rasterized polygons. Use either this or `raster_values_column`."
+        )
+        raster_values_column: Optional[str] = Field(
             description=(
-                "Parameters used by self.preprocess_dataset method. If set to `None` it skips the dataframe preprocess"
-                " step."
+                "GeoPandas dataframe column from which to read rasterization values for each geometry. Use either this"
+                " or `raster_value`."
             )
         )
+        output_feature: Feature = Field(description="A feature which should contain the newly rasterized data.")
+        polygon_buffer: float = Field(
+            0, description="The size of polygon buffering to be applied before rasterization."
+        )
+        resolution: Optional[float] = Field(
+            description="Rendering resolution in meters. Cannot be used with `raster_shape`."
+        )
+        raster_shape: Optional[Tuple[int, int]] = Field(
+            description="Shape of resulting raster image. Cannot be used with `resolution`."
+        )
+        overlap_value: Optional[float] = Field(description="Value to write over the areas where polygons overlap.")
+        dtype: np.dtype = Field(np.dtype("int32"), description="Numpy dtype of the output feature.")
+        no_data_value: int = Field(0, description="The no_data_value argument to be passed to VectorToRasterTask")
+
+        preprocess_dataset: Optional[Preprocessing] = Field(
+            description="Parameters used by `self.preprocess_dataset` method. Skipped if set to `None`."
+        )
         compress_level: int = Field(1, description="Level of compression used in saving EOPatches")
 
+        _parse_dtype = field_validator("dtype", parse_dtype, pre=True)
+        _check_raster_value_values_column = ensure_exactly_one_defined("raster_value", "raster_values_column")
+        _check_shape_resolution = ensure_exactly_one_defined("raster_shape", "resolution")
+        _check_output_feature_type = field_validator("output_feature", restrict_types(lambda ftype: ftype.is_image()))
+
+        @validator("vector_input")
+        def _check_vector_input(cls, vector_input: Union[Feature, str]) -> Union[Feature, str]:
+            if isinstance(vector_input, str):
+                if not vector_input.lower().endswith((".geojson", ".shp", ".gpkg", ".gdb")):
+                    raise ValueError(
+                        f"Input file path {vector_input} should be a GeoJSON, Shapefile, GeoPackage or GeoDataBase."
+                    )
+            else:
+                parse_feature(feature=vector_input, allowed_feature_types=lambda ftype: ftype.is_vector())
+            return vector_input
+
+        @validator("output_feature")
```

Comment:
if you already validate the output feature here, you could also add in an `assert raster_ftype.is_image()` and get rid of the additional validator.

Reply:
done